src/libraries/Microsoft.PowerFx.Interpreter/Functions/CollectFunction.cs

```diff
@@ -0,0 +1,267 @@
+            // document errors for invalid arguments such as unsupported aggregate types.
+            fValid &= TryGetUnifiedCollectedType(args, argTypes, errors, out DType collectedType);
+            Contracts.Assert(collectedType.IsTable);
+
+            // The item type must be compatible with the collection schema.
+            var fError = false;
+            returnType = DType.Union(ref fError, collectionType, collectedType, useLegacyDateTimeAccepts: true);
+            if (fError)
+            {
+                fValid = false;
+                if (!SetErrorForMismatchedColumns(collectionType, collectedType, args[1], errors))
+                {
+                    errors.EnsureError(DocumentErrorSeverity.Severe, args[0], TexlStrings.ErrNeedValidVariableName_Arg);
+                }
+            }
+
+            return fValid;
+        }
+
+        // This method returns true if there are special suggestions for a particular parameter of the function.
+        public override bool HasSuggestionsForParam(int argumentIndex)
+        {
+            Contracts.Assert(argumentIndex >= 0);
+
+            return argumentIndex == 0;
+        }
+
+        public override bool TryGetDataSourceNodes(CallNode callNode, TexlBinding binding, out IList<FirstNameNode> dsNodes)
+        {
+            Contracts.AssertValue(callNode);
+            Contracts.AssertValue(binding);
+
+            dsNodes = new List<FirstNameNode>();
+            if (callNode.Args.Count != 2)
+            {
+                return false;
+            }
+
+            var args = Contracts.VerifyValue(callNode.Args.Children);
+            var arg1 = Contracts.VerifyValue(args[1]);
+
+            // Only the second arg can contribute to the output for the purpose of delegation
+            return ArgValidators.DataSourceArgNodeValidator.TryGetValidValue(arg1, binding, out dsNodes);
+        }
+
+        public override IEnumerable<Identifier> GetIdentifierOfModifiedValue(TexlNode[] args, out TexlNode identifierNode)
+        {
+            Contracts.AssertValue(args);
+
+            identifierNode = null;
+            if (args.Length == 0)
+            {
+                return null;
+            }
+
+            var firstNameNode = args[0]?.AsFirstName();
+            identifierNode = firstNameNode;
+
+            if (firstNameNode == null)
+            {
+                return null;
+            }
+
+            var identifiers = new List<Identifier>
+            {
+                firstNameNode.Ident
+            };
+            return identifiers;
+        }
+
+        public override bool IsAsyncInvocation(CallNode callNode, TexlBinding binding)
+        {
+            Contracts.AssertValue(callNode);
+            Contracts.AssertValue(binding);
+
+            return Arg0RequiresAsync(callNode, binding);
+        }
+
+        public Task<FormulaValue> InvokeAsync(FormulaValue[] args, CancellationToken cancel)
```

Comment:
>public Task<FormulaValue> [](http://example.com/codeflow?start=8&length=25)

use async keyword.  The consistency is important here, especially because there are subtle diferences with exception handling. 

That will also simplify line 264.

Reply:
Done.